kratos/tests/cpp_tests/sources/test_kratos_parameters.cpp

```diff
@@ -0,0 +1,976 @@
+            "hi",
+            false
+        ],
+        "tmp": 5.0
+    },
+    "new_default_obj": {
+        "aaa": "string",
+        "bbb": false,
+        "ccc": 22
+    },
+    "new_default_value": -123.0,
+    "string_value": "hello"
+})";
+    return expected_validation_output;
+}
+
+std::string GetJSONStringFourLevels()
+{
+    const std::string four_levels = R"({
+        "bool_value": true,
+        "double_value": 2.0,
+        "int_value": 10,
+        "level1": {
+            "level2": {
+                "level3": {
+                    "level4": {
+                    }
+                }
+            }
+        },
+        "string_value": "hello"
+    })";
+    return four_levels;
+}
+
+std::string GetJSONStringForLevelsVariation()
+{
+    const std::string four_levels_variation = R"({
+        "bool_value": true,
+        "double_value": 2.0,
+        "int_value": 10,
+        "level1": {
+            "a":11.0,
+            "level2": {
+                "level3": {
+                    "level4": {
+                    }
+                }
+            }
+        },
+        "string_value": "hello"
+    })";
+    return four_levels_variation;
+}
+
+std::string GetJSONStringForLevelsWrongVariation()
+{
+    const std::string four_levels_wrong_variation = R"({
+        "int_value": 10,
+        "double_value": "hi",
+        "bool_value": true,
+        "string_value": "hello",
+        "level1": {
+            "a":11.0,
+            "level2": {
+                "level3": {
+                    "level4": {
+                    }
+                }
+            }
+        }
+    })";
+    return four_levels_wrong_variation;
+}
+
+std::string GetJSONStringForLevelsDefaults()
+{
+    const std::string four_levels_defaults = R"({
+        "bool_value": true,
+        "double_value": 2.0,
+        "int_value": 10,
+        "level1": {
+            "a":1.0,
+            "level2": {
+                "b":2.0,
+                "level3": {
+                    "c":3.0,
+                    "level4": {
+                        "d":4.0
+                    }
+                }
+            }
+        },
+        "string_value": "hello"
+    })";
+    return four_levels_defaults;
+}
+
+KRATOS_TEST_CASE_IN_SUITE(KratosParameters, KratosCoreFastSuite)
+{
+    Parameters kp = Parameters(GetJSONString());
+    KRATOS_CHECK_STRING_EQUAL(
+        kp.WriteJsonString(),
+        R"({"bool_value":true,"double_value":2.0,"int_value":10,"level1":{"list_value":[3,"hi",false],"tmp":5.0},"string_value":"hello"})"
+    );
+
+    KRATOS_CHECK(kp.Has("int_value"));
+    KRATOS_CHECK_IS_FALSE(kp.Has("unextisting_value"));
+
+    KRATOS_CHECK_EQUAL(kp["int_value"].GetInt(), 10);
+    KRATOS_CHECK_EQUAL(kp["double_value"].GetDouble(), 2.0);
+    KRATOS_CHECK_EQUAL(kp["bool_value"].GetBool(), true);
+    KRATOS_CHECK_EQUAL(kp["string_value"].GetString(), "hello");
+
+    KRATOS_CHECK_STRING_EQUAL(kp.PrettyPrintJsonString(), GetJSONStringPrettyOut());
```

Comment:
hm not sure abt this on, json is unordered, this might fail on a different machine
(assuming it always has the same order screwed me in the past)

maybe better add size check and checks for individual values like you already do

Reply:
In theory should work always (or the python tests would fail)